tests/providers/mysql/hooks/test_mysql.py

```diff
@@ -68,9 +68,83 @@ def test_get_uri(self, mock_connect):
         self.connection.extra = json.dumps({'charset': 'utf-8'})
         self.db_hook.get_conn()
         assert mock_connect.call_count == 1
-        args, kwargs = mock_connect.call_args
+        assert self.db_hook.get_uri() == "mysql://login:password@host/?charset=utf-8"
+
+    @mock.patch('MySQLdb.connect')
+    def test_get_uri_without_extra(self, mock_connect):
+        self.connection.schema = 'schema'
+        self.connection.port = '3306'
+        self.db_hook.get_conn()
+        assert mock_connect.call_count == 1
+        assert self.db_hook.get_uri() == "mysql://login:password@host:3306/schema"
+
+    @mock.patch('MySQLdb.connect')
+    def test_get_uri_with_schema(self, mock_connect):
+        self.connection.schema = 'schema'
+        self.connection.extra = json.dumps({'charset': 'utf-8'})
+        self.db_hook.get_conn()
+        assert mock_connect.call_count == 1
         assert self.db_hook.get_uri() == "mysql://login:password@host/schema?charset=utf-8"
 
+    @mock.patch('MySQLdb.connect')
+    def test_get_uri_with_port(self, mock_connect):
+        self.connection.port = '3306'
+        self.connection.extra = json.dumps({'charset': 'utf-8'})
+        self.db_hook.get_conn()
+        assert mock_connect.call_count == 1
+        assert self.db_hook.get_uri() == "mysql://login:password@host:3306/?charset=utf-8"
+
+    @mock.patch('MySQLdb.connect')
+    def test_get_uri_with_port_and_empty_host(self, mock_connect):
+        self.connection.host = None
+        self.connection.port = '3306'
+        self.connection.extra = json.dumps({'charset': 'utf-8'})
+        self.db_hook.get_conn()
+        assert mock_connect.call_count == 1
+        assert self.db_hook.get_uri() == "mysql://login:password@:3306/?charset=utf-8"
+
+    @mock.patch('MySQLdb.connect')
+    def test_get_uri_with_port_and_schema(self, mock_connect):
+        self.connection.port = '3306'
+        self.connection.schema = 'schema'
+        self.connection.extra = json.dumps({'charset': 'utf-8'})
+        self.db_hook.get_conn()
+        assert mock_connect.call_count == 1
+        assert self.db_hook.get_uri() == "mysql://login:password@host:3306/schema?charset=utf-8"
+
+    @mock.patch('MySQLdb.connect')
+    def test_get_uri_without_password(self, mock_connect):
+        self.connection.port = '3306'
+        self.connection.schema = 'schema'
+        self.connection._password = None
```

Comment:
Isn't it better to use set_password? airflow.models.Connection contains this method to work with _password field

Reply:
@wojsamjan I want to make `_password` empty. `set_password` is not suitable to this, because under the hood this method checks the value for None.